src/jit/morph.cpp

```diff
@@ -18154,6 +18127,14 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
             MorphStructField(node, user);
         }
         else if (node->OperIs(GT_LCL_VAR, GT_LCL_FLD))
+        {
+            if (node->OperIs(GT_LCL_FLD))
+            {
+                MorphLocalField(node, user);
+            }
+        }
+
+        if (node->OperIsLocal())
```

Comment:
It changes the condition a bit, from `node->OperIs(GT_LCL_VAR, GT_LCL_FLD, GT_FIELD)` to `node->OperIsLocal()` that includes `GT_PHI_ARG, GT_REG_VAR, GT_STORE_LCL_VAR, GT_STORE_LCL_FLD` as well. However, we do not expect any of the additional nodes in morph so it doesn't change the behavior.

Reply:
> GT_STORE_LCL_VAR, GT_STORE_LCL_FLD

Probably a good thing when we'll get rid of GT_ASG.